main.go

```diff
@@ -25,5 +25,13 @@ func mainWithError() error {
 	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 	defer stop()
 
-	return cmd.Run(ctx, os.Args, health.NewHealth(), nil)
+	args := os.Args[1:]
+
+	for _, p := range args {
```

Comment:
I'd suggest using cobra [1] instead of manually parsing command line arguments as it's happening now. We could also leverage sub-commands instead of relying on strict flag placement as this suggests.

[1] https://github.com/spf13/cobra

Reply:
i think so too. this is just a stop gap until we merge all the rsyslog code into main.

Reply:
do you think adding cobra to this PR would make this PR too large?

Reply:
I don't think so, it's not that much more code.

Reply:
sounds good. starting that right now.